xdsl/transforms/memref_stream_tile_outer_loops.py

```diff
@@ -0,0 +1,216 @@
+from collections.abc import Sequence
+from dataclasses import dataclass, field
+from typing import cast
+
+from xdsl.context import MLContext
+from xdsl.dialects import affine, arith, memref, memref_stream, scf
+from xdsl.dialects.builtin import (
+    AffineMapAttr,
+    ArrayAttr,
+    IndexType,
+    IntegerAttr,
+    MemRefType,
+    ModuleOp,
+    NoneAttr,
+    StridedLayoutAttr,
+)
+from xdsl.ir import Attribute, Block, Operation, Region, SSAValue
+from xdsl.ir.affine import AffineMap
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.rewriter import InsertPoint, Rewriter
+from xdsl.utils.exceptions import DiagnosticException
+
+
+def insert_subview(
+    memref_val: SSAValue,
+    affine_map: AffineMap,
+    apply_operands: Sequence[SSAValue],
+    upper_bounds: Sequence[int],
+    location: InsertPoint,
+):
+    """
+    A helper method to insert a subview from the input `memref_val` to one with new upper
+    bounds, given that it will be accessed with the specified affine map.
+    `apply_operands` are the operands to use to determine the new offset,
```

Comment:
```suggestion
    `offset` are the indices to use to determine the new offset,
```
The `apply_operands` name is an implementation detail. If it represents the offset then lets name it as such